Tests/mocks/rdkshell.h

```diff
@@ -0,0 +1,714 @@
+	  }
+          virtual void onApplicationSuspended(const std::string& client)
+	  {
+                 getInstance().impl->onApplicationSuspended(client);
+	  }
+          virtual void onApplicationResumed(const std::string& client)
+	  {
+		  getInstance().impl->onApplicationResumed(client);
+	  }
+          virtual void onApplicationActivated(const std::string& client)
+	  {
+		  getInstance().impl->onApplicationActivated(client);
+	  }
+          virtual void onUserInactive(const double minutes)
+	  {
+		  getInstance().impl->onUserInactive(minutes);
+	  }
+          virtual void onDeviceLowRamWarning(const int32_t freeKb, const int32_t availableKb, const int32_t usedSwapKb) 
+	  {
+		  getInstance().impl->onDeviceLowRamWarning(freeKb, availableKb, usedSwapKb);
+	  }
+          virtual void onDeviceCriticallyLowRamWarning(const int32_t freeKb, const int32_t availableKb, const int32_t usedSwapKb) 
+	  {
+		  getInstance().impl->onDeviceCriticallyLowRamWarning(freeKb, availableKb, usedSwapKb);
+	  }
+          virtual void onDeviceLowRamWarningCleared(const int32_t freeKb, const int32_t availableKb, const int32_t usedSwapKb)
+	  {
+		  getInstance().impl->onDeviceLowRamWarningCleared(freeKb, availableKb, usedSwapKb);
+	  }
+          virtual void onDeviceCriticallyLowRamWarningCleared(const int32_t freeKb, const int32_t availableKb, const int32_t usedSwapKb)
+	  {
+		  getInstance().impl->onDeviceCriticallyLowRamWarningCleared(freeKb, availableKb, usedSwapKb);
+	  }
+          virtual void onEasterEgg(const std::string& name, const std::string& actionJson)
+	  {
+		  getInstance().impl->onEasterEgg(name, actionJson);
+          }
+          virtual void onPowerKey() 
+	  {
+		  getInstance().impl->onPowerKey();
+	  }
+          virtual void onSizeChangeComplete(const std::string& client) 
+	  {
+		 getInstance().impl->onSizeChangeComplete(client);
+	  }
+	  const std::string RDKSHELL_EVENT_SIZE_CHANGE_COMPLETE = "onSizeChangeComplete";
+    };
+
+
+   struct RdkShellEasterEggKeyDetails
+    {
+        RdkShellEasterEggKeyDetails() {}
+        RdkShellEasterEggKeyDetails(uint32_t code, uint32_t modifiers, uint32_t holdTime): keyCode(code), keyModifiers(modifiers), keyHoldTime(holdTime) {}
+        uint32_t keyCode;
+        uint32_t keyModifiers;
+        uint32_t keyHoldTime;
+    };
+
+    struct RdkShellEasterEggDetails
+    {
+        std::string id;
+        std::vector<RdkShellEasterEggKeyDetails> keySequence;
+        uint32_t duration;
+        std::string api;
+    };
+
+    class EasterEgg;
+
+
+    class RdkShellApiImpl
+    {
+            public:
+                    virtual bool systemRam(uint32_t& freeKb, uint32_t& totalKb, uint32_t& availableKb, uint32_t& usedSwapKb) = 0;
+                    virtual void setMemoryMonitor(std::map<std::string, RdkShellData> &configuration) = 0;
+                    virtual void enableFlushing(bool enable) = 0;
+                    virtual bool isFlushingEnabled() = 0;
```

Comment:
enableFlushing and isFlushingEnabled belong to Logger so should be in LoggerImpl for the separation of concern